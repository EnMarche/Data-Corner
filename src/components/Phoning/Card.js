import PropTypes from 'prop-types'
import { Card, CardActions, CardContent, CardHeader } from '@mui/material'

const UICard = ({
  headerTitle,
  headerSubtitle,
  content,
  actions,
  rootProps = {},
  headerProps = {},
  contentProps = {},
  actionsProps = {},
}) => (
  <Card {...rootProps}>
    {headerTitle && <CardHeader {...headerProps} title={headerTitle} subheader={headerSubtitle} />}

    <CardContent sx={{ ...(contentProps.sx || {}), py: 0 }} {...contentProps}>
      {content}
    </CardContent>

    {actions && <CardActions {...actionsProps}>{actions}</CardActions>}
  </Card>
)

UICard.propTypes = {
  headerTitle: PropTypes.node,
  headerSubtitle: PropTypes.node,
  content: PropTypes.node,
  actions: PropTypes.node,
  rootProps: PropTypes.object,
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  actionsProps: PropTypes.object,
}

export default UICard
