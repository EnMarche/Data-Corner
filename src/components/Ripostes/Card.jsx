/* eslint-disable react/forbid-prop-types */
import React from 'react';
import {
    makeStyles, createStyles, Grid, Paper, Button, Box,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import DotMenu from '../HelperComponents/DotMenu';

const useStyles = makeStyles((theme) => createStyles({
    root: {
        padding: '16px',
        borderRadius: '8.35px',
    },
    container: {
        marginBottom: '8px',
    },
    chip: {
        fontSize: '10px',
        fontWeight: '500',
        borderRadius: '19px',
        padding: '2px 8px',
    },
    withBorder: {
        border: `1px solid ${theme.palette.gray200}`,
    },
    active: {
        color: theme.palette.teal700,
        background: theme.palette.activeLabel,
        borderRadius: '19px',

    },
    inactive: {
        color: theme.palette.red600,
        background: theme.palette.inactiveLabel,
    },
    date: {
        fontSize: '10px',
        color: theme.palette.gray600,
        padding: '7px 8px',
    },
    title: {
        fontSize: '16px',
        fontWeight: '600',
        color: theme.palette.gray900,
        width: '400px',
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    },
    creator: {
        fontSize: '12px',
        fontWeight: '400',
        color: theme.palette.gray600,
    },
    KpiContainer: {
        marginBottom: '20px',
    },
    buttonContainer: {
        position: 'relative',
        bottom: '0',
    },
    editButton: {
        fontSize: '13px',
        fontWeight: '500',
        color: theme.palette.teal700,
        marginTop: '11px',
        '&:hover': {
            background: theme.palette.riposteBackground,
            borderRadius: '8.35px',
        },
    },
}));

const Card = ({
    item, handleClickOpen, handleActiveItem,
}) => {
    const classes = useStyles();

    return (
        <Grid item xs={12} sm={6} md={4}>
            <Paper classes={{ root: classes.root }}>
                <Grid container className={classes.container}>
                    <Grid item>
                        {item.enabled ? <span className={`${classes.chip} ${classes.active}`}>Active</span> : <span className={`${classes.chip} ${classes.inactive}`}>Inactive</span>}
                    </Grid>
                    <Grid item className={classes.date}>
                        Le {new Date(item.created_at).toLocaleDateString()}
                    </Grid>
                </Grid>
                <Grid container className={classes.container}>
                    <Grid item className={classes.title}>{item.title}</Grid>
                    <Grid item className={classes.creator}>{item.creator}</Grid>
                </Grid>
                <Grid container spacing={1} className={classes.KpiContainer}>
                    <Grid item>
                        {item.with_notification ? <Box className={`${classes.chip} ${classes.withBorder}`}><i className="fas fa-bell" /></Box> : <Box className={`${classes.chip} ${classes.withBorder}`}><i className="fas fa-bell-slash" /></Box>}
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_views} vue{item.nb_views > 1 && 's'}
                        </Box>
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_detail_views} vue{item.nb_detail_views > 1 && 's'} détaillée{item.nb_detail_views > 1 && 's'}
                        </Box>
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_ripostes} riposte{item.nb_ripostes > 1 && 's'}
                        </Box>
                    </Grid>
                </Grid>
                <Grid container justifyContent="space-between" className={classes.buttonContainer}>
                    <Grid item>
                        <Button
                            className={classes.editButton}
                            onClick={() => handleClickOpen(item.uuid)}
                        >Éditer
                        </Button>
                    </Grid>
                    <Grid item>
                        <DotMenu handleActiveItem={handleActiveItem} item={item} />
                    </Grid>
                </Grid>
            </Paper>
        </Grid>
    );
};

export default Card;

Card.propTypes = {
    item: PropTypes.object.isRequired,
    handleClickOpen: PropTypes.func.isRequired,
    handleActiveItem: PropTypes.func.isRequired,
};
