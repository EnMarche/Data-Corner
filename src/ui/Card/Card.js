import PropTypes from 'prop-types'
import { Paper } from '@mui/material'
import { styled } from '@mui/system'

import { VerticalContainer } from 'components/shared/styled'

export const Root = styled(Paper)(
  ({ theme }) => `
  padding: ${theme.spacing(0, 2, 2, 2)};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-radius: 8px;
  box-shadow: 0px 0px 5px rgba(29, 33, 79, 0.03);
  &:hover {
    box-shadow: 0px 4px 10px rgba(29, 33, 79, 0.06);
  }
`
)

const UICard = ({
  header,
  content,
  actions,
  rootProps = {},
  headerProps = {},
  contentProps = {},
  actionsProps = {},
}) => (
  <Root {...rootProps} data-cy="ui-card">
    {header && <VerticalContainer {...headerProps}>{header}</VerticalContainer>}
    {content && <VerticalContainer {...contentProps}>{content}</VerticalContainer>}
    {actions && <VerticalContainer {...actionsProps}>{actions}</VerticalContainer>}
  </Root>
)

UICard.propTypes = {
  header: PropTypes.node,
  content: PropTypes.node,
  actions: PropTypes.node,
  rootProps: PropTypes.object,
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  actionsProps: PropTypes.object,
}

export default UICard
