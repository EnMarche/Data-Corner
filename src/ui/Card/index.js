export { default } from './Card'

export { Title } from './Title/Title'
export { Chip } from './Chip/Chip'
export { CtaButton } from './CtaButton/CtaButton'
