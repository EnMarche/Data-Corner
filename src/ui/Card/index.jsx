export { default } from './Card'

export { Title } from './Title/Title'
export { UIChip } from './Chip/Chip'
export { CtaButton } from './CtaButton/CtaButton'
