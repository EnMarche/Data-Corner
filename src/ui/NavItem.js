import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { Icon, Typography } from '@mui/material'

import { makeStyles } from '@mui/styles'

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    marginLeft: theme.spacing(4),
    padding: theme.spacing(1.5, 2),
    borderRadius: '8px 0 0 8px',
    color: theme.palette.gray600,
    '&:hover': {
      color: theme.palette.gray600,
      background: theme.palette.gray100,
    },
  },
  active: {
    color: ({ color }) => `${theme.palette[color] || 'initial'}`,
    background: ({ bgColor }) => bgColor || 'transparent',
    '&:hover': {
      color: ({ color }) => `${theme.palette[color] || 'initial'}`,
      background: ({ bgColor }) => bgColor || 'transparent',
    },
  },
  icon: {
    fontSize: '14px',
    paddingRight: theme.spacing(2),
  },
}))

const UINavItem = ({ path, label, icon = null, color = null, bgColor = null }) => {
  const classes = useStyles({ color, bgColor })

  return (
    <NavLink className={({ isActive }) => [].concat(classes.root, isActive ? classes.active : '').join(' ')} to={path}>
      <Icon component={icon} className={classes.icon} />
      <Typography variant="body2">{label}</Typography>
    </NavLink>
  )
}

UINavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.elementType,
  color: PropTypes.string,
  bgColor: PropTypes.string,
}

export default UINavItem
