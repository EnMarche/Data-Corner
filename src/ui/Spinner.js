import { CircularProgress, Grid } from '@mui/material'

const spinerMessages = {
  loading: 'Page en cours de chargement',
}
const Spinner = () => (
  <Grid container style={{ marginTop: 'calc(100vh - 70vh)', textAlign: 'center' }}>
    <Grid item xs={12}>
      <CircularProgress sx={{ color: '#0049C6' }} />
    </Grid>
    <Grid item xs={12}>
      <strong>{spinerMessages.loading}</strong>
    </Grid>
  </Grid>
)

export default Spinner
