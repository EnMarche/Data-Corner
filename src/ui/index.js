export { default as UINavItem } from './NavItem';
export { default as Icons } from './constants/icons';
export { default as Colors } from './constants/colors';
