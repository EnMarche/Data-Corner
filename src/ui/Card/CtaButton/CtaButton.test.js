import { render } from '@testing-library/react'
import { CtaButton } from 'ui/Card'

jest.mock('@mui/system', () => ({
  styled: c => () => c,
}))
jest.mock('@mui/icons-material', () => {
  const icons = {
    __esModule: true,
  }

  const handler = {
    get: function (_, prop) {
      return () => <div className={`mock_${prop}Icon`} />
    },
  }

  return new Proxy(icons, handler)
})
jest.mock('@mui/material', () => ({
  Button: ({ children, color }) => (
    <div className="mui-button-mock" data-color={color}>
      {children}
    </div>
  ),
}))
describe('CtaButton', () => {
  it('displays a CtaButton', () => {
    const { container } = render(<CtaButton color="color">foo</CtaButton>)
    expect(container).toMatchSnapshot()
  })
})
