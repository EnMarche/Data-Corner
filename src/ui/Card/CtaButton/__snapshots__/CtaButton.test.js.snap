// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButton displays a CtaButton 1`] = `
<div>
  <div
    class="mui-button-mock"
  >
    foo
  </div>
</div>
`;
